### ISSUE TITLE
Relative paths don't work for template?

### ISSUE THREAD

**Issue by YouveGotMeowxy** [sebastianfeldmann/phpbu#364]
Am I makin ya nuts yet, @sebastianfeldmann ? LOL :)

Not super important, but wondering why this happens? If I use the full server path to the template file, PHPBU finds the template file w/out complaint; but if I use the relative path it complains that it can't find it?

However, I use the exact same relative path for the .log file, and BU has no problem finding it. Both the template file and the .log file are in the same folder as the BU script.

I can use the full path to point to the template, but for the sake of less clutter, and readability I like to use relative paths sometimes.

![image](https://github.com/sebastianfeldmann/phpbu/assets/39491297/ac850a05-9fbc-4f1b-bad1-84477ba3999c)

**Comment by sebastianfeldmann** [comment]
Normally the path should be relative to the configuration file.

There is currently a bug that for the webhook template that is not the case.

**Comment by sebastianfeldmann** [comment]
Will be fixed in the upcoming version

**Comment by sebastianfeldmann** [comment]
So `phpbu` is looking at path relative to the **configuration** file. 
This is done because it should not matter where you are executing `phpbu` as long as you execute a specific configuration file.

So cwd being `/` and calling `phpbu --configuration=/foo/bar/phpbu.xml` will assume all relative path within that configuration are relatively to that configuration file. Writing the to a log file located in `/foo/bar/baz/json.log` can be referenced as `baz/json.log`.

This should be true for all path configurations within the whole configuration. Absolute or relative to the configuration file.

If you encounter any issues, where it does not work like that feel free to report it and I will look into the specifics

**Comment by YouveGotMeowxy** [comment]
Heyas @sebastianfeldmann , glad to see yur still alive n kickin. I was starting to get worried. ;P

Regarding my original screenshot, since all files are in the same folder - the bu file, the config files, the log, etc., and `./` means the current folder, that means the current folder and relative folder to the config file, are all in the same place., no? I just woke up so maybe my brain isn't firing on all cylinders yet.

**Comment by sebastianfeldmann** [comment]
I don't see the `phpbu.xml` configuration file in that folder.

**Comment by YouveGotMeowxy** [comment]
> I don't see the `phpbu.xml` configuration file in that folder.

it's the file name that you can see at the top left titlebar in notepad, in the screenshot. it's a json file rather than xml though.

**Comment by YouveGotMeowxy** [comment]
> Ah my fault. You are using a `json` configuration file.

yea, I don't like xml. :p

**Comment by sebastianfeldmann** [comment]
This config works for me.

```xml
  <logging>
    <log type="json" target="backup/json.log"/>
  </logging>
```

This as well

```xml
    <backup name="tar">
      <source type="tar">
        <option name="path" value="./../crap"/>
```

Are you sure the log options does not work?

I will check the `webhook-template` option.

**Comment by YouveGotMeowxy** [comment]
It's the apprise file's relative path that seemed broken, not the log file's relative. path. It's all in the screenshot. :)

**Comment by sebastianfeldmann** [comment]
This works

```xml
    <log type="webhook">
        <option name="uri" value="https://my-webhook-url.com" />
        <option name="contentType" value="application/json" />
        <option name="method" value="post" />
        <option name="template" value="backup/webhook-tpl.json" />
    </log>
```

This fails expectedly (directory does not exist) with the same error message you encounter

template not found: /phpbu/phpbu/_____backup/webhook-tpl.json

```xml
    <log type="webhook">
        <option name="uri" value="https://my-webhook-url.com" />
        <option name="contentType" value="application/json" />
        <option name="method" value="post" />
        <option name="template" value="_____backup/webhook-tpl.json" />
    </log>
```

**Comment by sebastianfeldmann** [comment]
Maybe it is because you are using `phpbu` 5.2 :/ Maybe there is a problem in that version

**Comment by sebastianfeldmann** [comment]
Yes it is, the path resolution in 5.2 works differently for the `template` option.
It was "fixed" with 455827f1f6223a22e3cb280901f5b9829d0e7a3a

**Comment by sebastianfeldmann** [comment]
So this is already fixed, all you need to do is upgrade to the latest version :)

**Comment by YouveGotMeowxy** [comment]
> So this is already fixed, all you need to do is upgrade to the latest version :)

Ahh excellent :) I've been using version phpbu 6.0.28 for while now.

This is interesting though, but maybe phpbu 6.0.29 will also fix that? I just now noticed this.

![Image](https://github.com/user-attachments/assets/b3d393a7-2e51-4adf-a582-4110c51e5f27)

It looks like the config file was looking relative to the bu itself, rather than the config file, since I have the bu file 1 folder up, and configs in a subdir.

**Comment by YouveGotMeowxy** [comment]
oops, I was wrong on that last post. I deleted it.

**Comment by YouveGotMeowxy** [comment]
While I have your ear :p

is there any way to beef up the cleaning function to work with DropBox? I backup a lot of  DBs nightly and my DropBox is getting filled real fast. Like get a listing of files in the DropBox backup folder, and delete the ones no longer matching the rules?
